### PR TITLE
FIP 0010: describe proof spam attack

### PR DESCRIPTION
**Pull request by Stebalien**
This patch describes a "spam" attack with faulty proofs and explains why it's unlikely to be rational.

### CODE REVIEW

File: FIPS/fip-0010.md [hunk 1]
```diff
@@ -385,9 +386,35 @@ proofs, and disputing their own proofs. However:
 
 1. This attack is rate-limited. It can only be used to extract one reward per active partition every
    two days.
-2. Another miner may dispute the proof first and steal the reward. This is especially true because
+3. Another miner may dispute the proof first and steal the reward. This is especially true because
```
Comment (anorth) [inline FIPS/fip-0010.md:389]: This numbering looks odd

File: FIPS/fip-0010.md [hunk 2]
```diff
@@ -385,9 +386,35 @@ proofs, and disputing their own proofs. However:
...
+
+Given that this miner will have less power than necessary to mine blocks, the "vesting rewards" will
+be empty. Therefore, disputes will pay out no rewards.
+
+Most rational miners will simply ignore these proofs as the target miner cannot pay out a
+reward. However, for the good of the network, some parties may wish to dispute any/all invalid
+proofs.
+
+Luckily, even though the miner in question will no _pay out_ rewards to the party that submitted the
+dispute, they still need to pay them eventually or abandon the miner. That is, unpaid rewards are
```
Comment (anorth) [inline FIPS/fip-0010.md:411]:
```suggestion
dispute, they still need to pay the penalty and rewards eventually or abandon the miner. That is, unpaid penalties and rewards are
```

**Comment by anorth** [comment]
Another possible path to a proof spam attack is a miner who has lots of power but has lost access to the sectors for some time and cannot submit a valid Window PoSt. 

If the sectors are gone for good, the miner choses between terminating the sectors immediately, or attempting vesting funds extraction before doing so. So this case resembles the previous one, but the angle here is a large volume of proofs to be verified. The miner will race other parties to submit the disputes, and hope to win some races to extract some funds. But I guess at equilibrium the miner should expect to lose most of those races except where they mine a block (because the disputes should be stolen by block producers).

If the sectors are not gone for good, the miner choses between missing the Window PoSts or submitting fakes until they can recover. So long as the penalty associated with a fraudulent proof exceeds that associated with a missed PoSt, and the miner expects to recover to vest their funds, there seems to be no incentive to pay the higher fraud penalty (which they can only choose to do once per partition anyway). Maybe the worst case here is that a large miner submits all fraudulent PoSts, consuming ~all the chain bandwidth to verify them, pushing the base fee up to (or past) the reward breakeven rate until they can all be processed.

I think it would be helpful for us to calculate the fraud proof rate that would saturate chain bandwidth, and hence the storage power associated with a deadline which, if fully fraudulent, would raise the base fee to this level for 30min. *This is an argument for keeping the dispute reward modest, rather than generous.* And maybe an argument for limiting maximum miner actor size.

**Comment by Stebalien** [comment]
> If the sectors are gone for good, the miner choses between terminating the sectors immediately, or attempting vesting funds extraction before doing so. So this case resembles the previous one, but the angle here is a large volume of proofs to be verified. The miner will race other parties to submit the disputes, and hope to win some races to extract some funds. But I guess at equilibrium the miner should expect to lose most of those races except where they mine a block (because the disputes should be stolen by block producers).

This should be very rate limited but I'll write it up. Basically, a miner can recover 4FIL per partition, losing 20FIL plus IPF in the process. Given _current_ network rewards (0.128FIL/TiB), 4FIL is less than half a days worth of block rewards for a full partition. 

1. They cannot repeat the same attack against the same partition because they'd need to have the sectors to actually recover the partition.
2. They must perform the attack _immediately_ on losing the sectors, because the attack becomes impossible once the chain marks them as faulty.

**Comment by Stebalien** [comment]
> They cannot repeat the same attack against the same partition because they'd need to have the sectors to actually recover the partition.

Well, they might have lost _most_ of their storage, but not all.

**Comment by jennijuju** [comment]
@Stebalien does this pr describes the final FIP and ready to merge?

**Comment by Stebalien** [comment]
Yes.

**Review by jennijuju — APPROVED** [review]

File: FIPS/fip-0010.md [hunk 3]
```diff
@@ -387,7 +388,33 @@ proofs, and disputing their own proofs. However:
...
+3. Start proving the sector with bad proofs.
+
+Given that this miner will have less power than necessary to mine blocks, the "vesting rewards" will
+be empty. Therefore, disputes will pay out no rewards.
+
+Most rational miners will simply ignore these proofs as the target miner cannot pay out a
+reward. However, for the good of the network, some parties may wish to dispute any/all invalid
+proofs.
+
+Luckily, even though the miner in question will no _pay out_ rewards to the party that submitted the
```
Comment (arajasek) [inline FIPS/fip-0010.md:410]:
```suggestion
Luckily, even though the miner in question will offer no _pay out_ rewards to the party that submitted the
```

**Review by arajasek — APPROVED** [review]
one nit

**Pull request merged**